packages/store-sync/src/dozer/getSnapshot.ts

```diff
@@ -0,0 +1,88 @@
+import { LogFilter, SyncFilter, TableQuery } from "./common";
+import { Hex } from "viem";
+import { StorageAdapterBlock, SyncFilter as LegacyLogFilter } from "../common";
+import { fetchRecordsSql } from "./fetchRecordsSql";
+import { recordToLog } from "../recordToLog";
+import { getSnapshot as getSnapshotLogs } from "../getSnapshot";
+import { bigIntMin, isDefined } from "@latticexyz/common/utils";
+
+export type GetSnapshotArgs = {
+  dozerUrl: string;
+  storeAddress: Hex;
+  filters?: SyncFilter[];
```

Comment:
would it better or worse to split this into two args: `filters` and `queries`?

Reply:
I kinda like the fact that `table` is shared by both so you can start simple by providing just the table (if you wanna fetch the entire table) and only add sql to the ones where it's needed later